2.0/m_opban.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* $ModAuthor: Sebastian Nielsen */
+/* $ModAuthorMail: sebastian@sebbe.eu */
+/* $ModDesc: Implements extban +b o: Prevents these persons from aquiring a privileged position */
+/* $ModDepends: core 2.0 */
+
+#include "inspircd.h"
+class ModuleOPBan : public Module
+{
+public:
+	void init()
+	{
+		Implementation eventlist[] = { I_OnRawMode, I_On005Numeric };
+		ServerInstance->Modules->Attach(eventlist, this, sizeof(eventlist)/sizeof(Implementation));
+	}
+  
+	Version GetVersion()
+	{
+		return Version("Implements extban +b o: Prevents these persons from aquiring a privileged position",VF_OPTCOMMON);
+	}
+
+	ModResult OnRawMode(User* user, Channel* chan, const char mode, const std::string &param, bool adding, int pcnt)
+	{
+		if (chan && IS_LOCAL(user) && !IS_OPER(user) && !ServerInstance->ULine(user->server)) {
+			Membership* transmitter = chan->GetUser(user);
+			if ((mode == 'b') && (param.length() > 2) && (param[0] == 'o') && (param[1] == ':') && (transmitter->modes.find('q') == std::string::npos))
```

Comment:
Rather than hardcoding to q it might be a good idea to have a config option that can take a rank at which users are allowed to remove the mode (see the customprefix ranks in modules.conf.example for the defaults).